Gems/ExpressionEvaluation/Code/Source/ElementInformationSerializer.inl

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/Memory.h>
+#include <AzCore/Serialization/AZStdAnyDataContainer.inl>
+#include <AzCore/Serialization/Json/BaseJsonSerializer.h>
+#include <AzCore/Serialization/Json/JsonSerialization.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <ExpressionEvaluation/ExpressionEngine/ExpressionTypes.h>
+
+namespace AZ
+{
+    class ElementInformationSerializer
+        : public BaseJsonSerializer
+    {
+
+    public:
+        AZ_RTTI(ElementInformationSerializer, "{B33E6AA9-C700-4E3D-857C-55F362AFE57A}", BaseJsonSerializer);
+        AZ_CLASS_ALLOCATOR_DECL;
+
+    private:
+        using ElementInformation = ExpressionEvaluation::ElementInformation;
+
+        static const char* EmptyAnyIdentifier;
```

Comment:
Prefer using `constexpr` and declaring the name here. It's also possible to use an `AZStd::string_view` instead of a `const char*` which will help with the comparison later on.